vassal-app/src/main/java/VASSAL/counters/SendToLocation.java

```diff
@@ -382,18 +382,14 @@ public Command myKeyEvent(KeyStroke stroke) {
     Point dest = null;
 
     // If we're about to move a Mat, establish the initial relative positions of all its "contents"
-    if (GameModule.getGameModule().isMatSupport() && !"".equals(outer.getProperty(Mat.MAT_NAME))) {
-      mat = (Mat) Decorator.getDecorator(outer, Mat.class);
-      if (mat != null) {
-        //BR// Should we ONLY take Cargo that are currently selected (in the KeyBuffer)?
-        contents = new ArrayList<>(mat.getContents());
-        offsets = new ArrayList<>();
-        final Point basePt = outer.getPosition();
-        for (final GamePiece piece : contents) {
-          final Point pt = piece.getPosition();
-          pt.x -= basePt.x;
-          pt.y -= basePt.y;
-          offsets.add(pt);
+    if (GameModule.getGameModule().isMatSupport()) {
+      final String matName = (String)outer.getProperty(Mat.MAT_NAME);
+      if (matName != null && !"".equals(matName)) {
+        mat = (Mat) Decorator.getDecorator(outer, Mat.class);
+        if (mat != null) {
+          final Point basePt = outer.getPosition();
+          contents = new ArrayList<>(mat.getContents());
+          offsets = new ArrayList<>(mat.getOffsets(basePt.x, basePt.y));
```

Comment:
Do you need to make another copy of the offsets list here? `getOffsets` is already returning you one that it created locally.

Reply:
Oh so I should really just do...

```
contents = mat.getContents();
offsets = mat.getOffsets(basePt.x, basePt.y);
```

Yeah?

Reply:
If `getContents()` is also returning something that's a local, rather than a member of `mat`, then yeah, you don't need to make additional copies.